docs/proposals/kubemark.md

```diff
@@ -0,0 +1,163 @@
+on Master, which adds additional load, but they're not the part of default setup, so in the first version we won't simulate this behavior.
+
+In the first version we’ll assume that all started Pods will run indefinitely if not explicitly deleted. In the future we can add a model
+of short-running batch jobs, but in the initial version we’ll assume only serving-like Pods.
+
+### Heapster
+
+In addition to system components we run Heapster as a part of cluster monitoring setup. Heapster currently watches Events, Pods and Nodes through the API server. In the test setup we can use real heapster for watching API server, with mocked out piece that scrapes cAdvisor data from Kubelets.
+
+### Elasticsearch
+
+Similarly to Heapster Elasticsearch runs outside the Master machine but generates some traffic on it. Fluentd “daemon” running on Master periodically sends Docker logs it gathered to the Elasticsearch running on one of the Nodes. In the initial version we omit Elasticsearch, as it produces only a constant small load on Master Node that does not change with the size of the cluster.
+
+## Necessary work
+
+There are three more or less independent things that needs to be worked on:
+- HollowNode implementation, creating a library/binary that will be able to listen to Watches and respond in a correct fashion with Status updates. This also involves creation
+of a CloudProvider that can produce such Hollow Nodes.
+- Kubemark setup, including figuring networking model, number of Hollow Nodes that will be allowed to run on a single “machine”, writing setup/run/teardown scripts
+- Creating a Player component that will send requests to the API server putting a load on a cluster. This involves creating a way to specify desired workload. This task is
+very well isolated from the rest, as it is about sending requests to the real API server. Because of that we can discuss requirements separately.
+
+## Concerns
+
+Network performance most likely won't be a problem for the initial version if running on bare metal or 'pure' cloud provider, as Kubemark
+will be running on standard networking stack (no cloud-provider software routes, or overlay network is needed, as we don't need custom
+routing between Pods). Similarly we don't think that running Kubemark on Kubernetes virtualized cluster networking will cause noticable
+performance impact, but it requires testing.
+
+On the other hand when adding additional features it may turn out that we need to simulate Kubernetes Pod network. In such, when running
+'pure' Kubemark we may try one of the following:
+  - running overlay network like Flannel or OVS instead of using cloud providers routes,
+  - write simple network multiplexer to multiplex communications from the Hollow Kubelets/KubeProxies on the machine.
+
+In case of Kubemark on Kubernetes it may turn that we run into a problem with adding yet another layer of network virtualization, but we
+don't need to solve this problem now.
+
+## Work plan
```

Comment:
It made more sense when there was no "Concerns" in between:/ I don't have a good idea on what to do with those.